tools/alive-tv.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2018-present The Alive2 Authors.
+// Distributed under the MIT license that can be found in the LICENSE file.
+
+#include "llvm/ADT/StringExtras.h"
+#include "llvm/Bitcode/BitcodeReader.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/ManagedStatic.h"
+#include "llvm/Support/PrettyStackTrace.h"
+#include "llvm/Support/Signals.h"
+#include "llvm/Support/SourceMgr.h"
+#include "llvm/Support/ToolOutputFile.h"
+
+#include "ir/function.h"
+#include "llvm_util/utils.h"
+#include "smt/smt.h"
+#include "smt/solver.h"
+#include "tools/alive_parser.h"
+#include "util/config.h"
+#include "util/file.h"
+#include <cstdlib>
+#include <iostream>
+#include <optional>
+#include <string_view>
+#include <vector>
+
+using namespace IR;
+using namespace tools;
+using namespace util;
+using namespace std;
+using namespace llvm_util;
+
+static llvm::cl::opt<std::string>
+opt_file1(llvm::cl::Positional, llvm::cl::desc("first_bitcode_file"),
+    llvm::cl::Required, llvm::cl::value_desc("filename"));
+
+static llvm::cl::opt<std::string>
+opt_file2(llvm::cl::Positional, llvm::cl::desc("second_bitcode_file"),
+    llvm::cl::Required, llvm::cl::value_desc("filename"));
+
+static llvm::cl::opt<bool> opt_disable_undef("disable-undef-input",
+    llvm::cl::init(false),
+    llvm::cl::desc("Alive: Assume inputs are not undef (default=false)"));
+
+static llvm::cl::opt<bool> opt_disable_poison("disable-poison-input",
+    llvm::cl::init(false),
+    llvm::cl::desc("Alive: Assume inputs are not poison (default=false)"));
+
+static llvm::cl::opt<unsigned> opt_smt_to(
+  "tv-smt-to", llvm::cl::desc("Alive: timeout for SMT queries (default=1000)"),
+  llvm::cl::init(1000), llvm::cl::value_desc("ms"));
+
+static llvm::ExitOnError ExitOnErr;
+
+// adapted from llvm-dis.cpp
+static std::unique_ptr<llvm::Module> openInputFile(llvm::LLVMContext &Context,
+                                                   std::string InputFilename) {
+  std::unique_ptr<llvm::MemoryBuffer> MB =
+    ExitOnErr(errorOrToExpected(llvm::MemoryBuffer::getFile(InputFilename)));
+  std::unique_ptr<llvm::Module> M =
+    ExitOnErr(getOwningLazyBitcodeModule(std::move(MB), Context,
```

Comment:
Hmm, is it too much to ask for it to just natively handle the `.ll`, too (or it does already) ? :)
I kinda expect to be heavily used, much like the `alive` with sudo-ir

Reply:
do you have a pointer to a utility function that handles .bc and .ll? this must exist somewhere...

Reply:
let's leave this for now, it's not pressing and easy to add later